homeassistant/components/tradfri/__init__.py

```diff
@@ -137,10 +139,15 @@ async def async_keep_alive(now: datetime) -> None:
         if hass.is_stopping:
             return
 
+        gw_status = True
         try:
             await api(gateway.get_gateway_info())
         except RequestError:
             _LOGGER.error("Keep-alive failed")
+            gw_status = False
+
+        for entity in tradfri_data[ENTITIES]:
+            entity.set_hub_available(gw_status)
```

Comment:
Please use our dispatch helper to send a signal to the entities instead. Don't store the entities in a shared container. Connect the entity to the signal in `async_added_to_hass`. See eg:

https://github.com/home-assistant/core/blob/8dd7e4a39b096cedc35d4c4c7df2e197ba9b9c0e/homeassistant/components/ambient_station/__init__.py#L168

https://github.com/home-assistant/core/blob/8dd7e4a39b096cedc35d4c4c7df2e197ba9b9c0e/homeassistant/components/ambient_station/__init__.py#L257-L261

Reply:
Changed, a much better solution thanks.